tests/unit/test_salesforce.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+import pytest
+
+from viadot.sources import Salesforce
+
+TABLE_TO_DOWNLOAD = "Account"
+TABLE_TO_UPSERT = "Contact"
+TEST_LAST_NAME = "prefect-viadot-test"
+ID_TO_UPSERT = "0035E00001YGWK3QAP"
+
+
+@pytest.fixture(scope="session")
+def salesforce():
+    s = Salesforce(config_key="sales_force_dev")
+    yield s
+
+
+@pytest.fixture(scope="session")
+def test_df_data(salesforce):
+    data = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": [TEST_LAST_NAME],
+    }
+    df = pd.DataFrame(data=data)
+
+    yield df
+
+    data_restored = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": ["LastName"],
+    }
+    df_restored = pd.DataFrame(data=data_restored)
+    salesforce.upsert(df=df_restored, table=TABLE_TO_UPSERT)
+
+
+@pytest.fixture(scope="session")
+def test_df_external(salesforce):
+    data = {
+        "LastName": [TEST_LAST_NAME],
+        "SAPContactId__c": ["111"],
+    }
+    df = pd.DataFrame(data=data)
+    yield df
+
+    data_restored = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": ["LastName"],
+    }
+    df_restored = pd.DataFrame(data=data_restored)
+    salesforce.upsert(df=df_restored, table=TABLE_TO_UPSERT)
+
+
+def test_upsert_empty(salesforce):
+    try:
+        df = pd.DataFrame()
+        salesforce.upsert(df=df, table=TABLE_TO_UPSERT)
+    except Exception as exception:
+        assert False, exception
+
+
+def test_upsert_external_id_correct(salesforce, test_df_external):
```

Comment:
It seems like this only tests the insert part, and not the update part?

Reply:
This test only performs an update. For some reason I couldn't insert on records

Reply:
I corrected the structure of this test
[✅ 🎨Improved structure of upsert test](https://github.com/dyvenia/viadot/pull/629/commits/de73e4ca66fa277fb67d44c7cd5221f0c9eacc00)

Reply:
You're saying that this function doesn't work? Then this test should be failing. The upsert test needs to test two things:
- new records are inserted
- existing records are updated

This should also work on multiple records (`df` can have eg. 100 rows), not on a single row as you suggerst [here](https://github.com/dyvenia/viadot/blob/a3fea0e7499287c554de2babe2f6367de9a8f058/viadot/sources/salesforce.py#L81).

Reply:
I renamed the column and added a test inserting two rows using the upsert method.

[🎨Rename external_id var to external_id_column](https://github.com/dyvenia/viadot/pull/629/commits/b6e311e43d8a83f64c00a58fcc17fc4c52db9140)

[✅ Added inserting two rows](https://github.com/dyvenia/viadot/pull/629/commits/c8bdf4e81fad33033ce2bf5fee2f7aa48e5355cb)